### ISSUE TITLE
Error pulling image (latest) from base, image ID '27cf784147099545' is invalid 

### ISSUE THREAD

**Issue by pwaller** [docker/docker#9775]
Docker 1.4.1.

~~I have just developed on #9774, so it's very possible that I did something that messed up my installation~~ (I've reproduced elsewhere). However, I have completely wiped `/var/lib/docker` and tried from fresh and I still get this problem:

```
$ docker run base echo hi
Unable to find image 'base:latest' locally
Pulling repository base
b750fe79269d: Error pulling image (latest) from base, image ID '27cf784147099545' is invalid 
27cf78414709: Error downloading dependent layers 
FATA[0003] Error pulling image (latest) from base, image ID '27cf784147099545' is invalid
```

I instrumented [the docker source here](https://github.com/docker/docker/blob/fa801358d6c533d195dc8edc3f0dc8be5585dded/utils/utils.go#L198) with a log message to verify that it's *really* where the error message is coming from, and sure enough it is.

So is it possible this is a new bug? This behaviour was changed very recently (25 days ago): https://github.com/docker/docker/commit/bff1d9dbce76bed1e267a067eb4a1a74ef4da312

**Comment by pwaller** [comment]
Ping @crosbymichael, this one means that lots of instructions on the internet are broken for anyone who tries to build an image from base or tries a tutorial which tells them to say `docker run base <foo>`. I keep encountering this and I have to remove base from lots of places. I'm a bit surprised this has happened and few people seem to be noticing..

**Comment by xylifyx** [comment]
Confirmed in OpenSUSE thumbleweed, docker 1.4.0

   xxx@suseee:~/ubuntu> sudo docker pull base
   Pulling repository base
   b750fe79269d: Error pulling image (latest) from base, image ID '27cf784147099545' is invalid v1/, image ID '27cf784147099545' is invalid 
   FATA[0004] Error pulling image (latest) from base, image ID '27cf784147099545' is invalid 
   xxx@suseee:~/ubuntu> docker --version
   Docker version 1.4.0, build 4595d4f

**Comment by pwaller** [comment]
I see the problem. [The base image has been deprecated](https://registry.hub.docker.com/_/base/). That's fine, except there is lots of stuff (such as docker tutorials!) in the wild that depends on it working :(. It also seems like a simple fix, the base images just need re-pushing, I would have thought.

Also summon @tianon in case he knows what's up.

**Comment by tianon** [comment]
IMO, any tutorial or document anywhere that mentions the use of `base` needs to be updated, and I would consider the very mention of that image a firm bug.  If it were entirely up to me, I'd have deleted it quite some time ago and just ripped that band-aid right off.

It hasn't been updated in ages, and won't be, especially since the name itself is so confusing.

**Comment by tianon** [comment]
To further drive home the point, a quote from the image's metadata on that page:
> Updated 1 year, 8 months ago

**Comment by tianon** [comment]
As if all that wasn't enough justification for not using this image anymore, it's Ubuntu 12.10, which is [officially EOL as of May 16, 2014](https://wiki.ubuntu.com/Releases#End_of_life).

**Comment by pwaller** [comment]
The problem is, there are a *lot* of things in the wild depending on it. If there was an announcement about base being deprecated, I missed it (if there was, can someone please link to it?).

[Google `docker run base` (estimates ~360,000)](https://www.google.co.uk/search?hl=en&q=%22docker%20run%20base%22#hl=en&q=%22docker+run+*+base%22)

Even though it's broken, lots of people are *still* using it. Given that I had not heard that the image was  dead, I think the message is unlikely to get through very easily.

I think we need to fix the problem that `docker run base` currently gives this cryptic fatal error. I don't mind how it is fixed, but it seems like a problem to me. It could be fixed by having docker check the name and give a more sensible messages, or just by re-pushing the existing image with the new digest format.

**Comment by pwaller** [comment]
I just realised that google returned results matching `docker run foo/centos-base` and all sorts of other things, so the 360,000 estimate is way off.

**Comment by thaJeztah** [comment]
@tianon 

> IMO, any tutorial or document anywhere that mentions the use of base needs to be updated, and I would consider the very mention of that image a firm bug.

Just noticed that all the API documentation uses `base:latest` in the [examples](https://github.com/docker/docker/blob/master/docs/sources/reference/api/docker_remote_api_v1.17.md#list-containers) :smile:

**Comment by tianon** [comment]
Gar, but it shouldn't be for a very long time now: #1914

**Comment by thaJeztah** [comment]
@tianon also made this as a suggestion; https://github.com/docker/docker/issues/10142

**Comment by pwaller** [comment]
So how do we fix all of the other broken tutorials on the internet? Can we make docker give a warning if you try to use the base image? This error is pretty insidious and gives docker a bad image (punintended) to newcomers.

**Comment by tianon** [comment]
Is the current warning better or worse than a 404 would be? :wink:

**Comment by pwaller** [comment]
It is worse, since it looks broken and breaks in one version of a docker and not another. But still not ideal since it doesn't tell the user what alternative action they need to take.

On 16 January 2015 17:55:38 GMT+00:00, Tianon Gravi <notifications@github.com> wrote:
>Is the current warning better or worse than a 404 would be? :wink:
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/docker/docker/issues/9775#issuecomment-70293275

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by Depicus** [comment]
I have to agree it is worst than a 404, as a new user the reason I came here is because I thought Docker was broken, if I hadn't found this page I might have given up on Docker thinking it was broken and not production ready. This kind of error should always produce some form of valuable error message, the current message does not.

**Comment by tianon** [comment]
```console
$ docker pull base
Pulling repository base
FATA[0001] Error: image library/base:latest not found   
```

**Comment by pwaller** [comment]
Unless it seems possible that anyone will consider making it a warning in docker, I guess that's as good as it gets and brings this issue to a close.

**Comment by mborho** [comment]
This issue is about "image ID '27cf784147099545' is invalid", right? This bug prevents me from removing any images from my disc, which is critical for me. 

Is there a possible workaround?

**Comment by pwaller** [comment]
@mborho what command are you trying to run when you hit that error? You might try deleting the base image directly.

**Comment by mborho** [comment]
@pwaller 'docker rmi IMAGE-ID' Where I can delete the base images? Somewhere under /var/lib/docker I guess, but where?

**Comment by pwaller** [comment]
@mborho I meant if you run `docker rmi <image id of the base image>`. But I assume you've tried that and it didn't work. Are you getting errors about the base image when you try to delete images which are not the base image?

**Comment by mborho** [comment]
@pwaller I got this error message when I've tried to delete *any* image, actually. Switched back to 1.3.2 now, but I'll install the latest version again and will try again.

**Comment by mborho** [comment]
@pwaller it fails at every "docker rmi" command with 1.4.1. The base-image on my hardrive is from Dec 2013. "27cf784147099545" is the only ID in /var/lib/docker/graph in the old short form. 

I can imagine, this bug will bite many people.

**Comment by mborho** [comment]
```go
validHex = regexp.MustCompile(`^([a-f0-9]{64})$`)
```
is causing this, see https://github.com/docker/docker/commit/acf1720b3f6035f0ab2c9658371ae8302851ffa3#diff-cd575165f84eed10d10eb1460b0927b4R35

**Comment by maxogden** [comment]
`rm -rf /var/lib/docker/graph/27cf784147099545` fixed this for me, thanks @mborho

**Comment by mcinnes01** [comment]
Hi 

This is my first go running docker and I using it in Hyper V running v 1.4.1, what is the work around to fix library/d:latest not found when I try to pull a repository e.g.

docker run –d –p 1880:1880 cpswan/node-red

Many thanks

Andy

**Comment by tiborvass** [comment]
Ping @shin- @ewindisch @tianon 

Did we (finally) remove `base`? I have `FATA[0005] Error: image library/base:latest not found`

**Comment by shin-** [comment]
I believe It had one of the legacy image IDs that we stopped supporting in 1.4+, and @tianon removed it as a result a few months ago.

tl;dr yes

**Comment by tiborvass** [comment]
Anybody who's using the extremely old `base` image, should rebuild them using `ubuntu`, really sorry for the inconvenience!
